zaza/openstack/charm_tests/ovn/tests.py

```diff
@@ -71,3 +76,231 @@ def setUpClass(cls):
+                'neutron-api-plugin-ovn',
+                'migrate-ovn-db',
+                action_params={
+                    'i-really-mean-it': True},
+                raise_on_failure=True,
+            )
+        )
+
+    # Charm readiness is no guarantee for API being ready to serve requests.
+    # https://bugs.launchpad.net/charm-neutron-api/+bug/1854518
+    @tenacity.retry(wait=tenacity.wait_exponential(min=5, max=60),
+                    reraise=True, stop=tenacity.stop_after_attempt(3))
+    def _run_migrate_mtu_action(self):
+        logging.info('Running `migrate-mtu` action on '
+                     'neutron-api-plugin-ovn/leader')
+        generic_utils.assertActionRanOK(
+            zaza.model.run_action_on_leader(
+                'neutron-api-plugin-ovn',
+                'migrate-mtu',
+                action_params={
+                    'i-really-mean-it': True},
+                raise_on_failure=True,
+            )
+        )
+
+    def _pause_units(self, application):
+        logging.info('Pausing {} units'.format(application))
+        zaza.model.run_action_on_units(
+            [unit.entity_id
+                for unit in zaza.model.get_units(application)],
+            'pause',
+            raise_on_failure=True,
+        )
+        self.target_deploy_status.update(
+            {
+                application: {
+                    'workload-status': 'maintenance',
+                    'workload-status-message': 'Paused',
+                },
+            },
+        )
+
+    def _run_cleanup_action(self):
+        logging.info('Running `cleanup` action on neutron-openvswitch units.')
+        zaza.model.run_action_on_units(
+            [unit.entity_id
+                for unit in zaza.model.get_units('neutron-openvswitch')],
+            'cleanup',
+            action_params={
+                'i-really-mean-it': True},
+            raise_on_failure=True,
+        )
+
+    def _resume_units(self, application):
+        logging.info('Resuming {} units'.format(application))
+        zaza.model.run_action_on_units(
+            [unit.entity_id
+                for unit in zaza.model.get_units(application)],
+            'resume',
+            raise_on_failure=True,
+        )
+        self.target_deploy_status.pop(application)
+
+    def test_ovs_ovn_migration(self):
+        """Test migration of existing Neutron ML2+OVS deployment to OVN.
+
+        The test should be run after deployment and validation of a legacy
+        deployment combined with subsequent run of a network connectivity test
+        on instances created prior to the migration.
+        """
+        # The setUp method of this test class will perform the migration steps.
+        # The tests.yaml is programmed to do further validation after the
+        # migration.
+
+        # Reset the n-gw and n-ovs instance-mtu configuration option so it does
+        # not influence how further tests are executed.
```

Comment:
If all the work is done prior to this, is this step necessary?

Reply:
Yes, after the migration is done we call [other tests](https://review.opendev.org/#/c/740974/18/tests/tests.yaml@63) to confirm that instances deployed prior to migration still have connectivity. That test would be influenced by the `instance-mtu` configuration option, and we don't want that.

Reply:
ACK